### PR TITLE
fix(sdk): add verify_ssl to kfp client, fixes #4569

### LINKED ISSUE #4569: KFP sdk client authentication error
/kind bug

**What steps did you take and what happened:**
Enabled authentication with Azure AD on AKS and installing Kubeflow with `kfctl_istio_dex.v1.1.0.yaml`  but skipping the dex from the manifest as Azure AD is an OIDC provider.  The load balancer is exposed over https with TLS 1.3 self-signed cert.

**OIDC Auth Service Configuration:**
 - client_id=XXXX
  - oidc_provider=https://login.microsoftonline.com/XXXX/v2.0
  - oidc_redirect_uri=https://XXXX/login/oidc
  - oidc_auth_url=https://login.microsoftonline.com/XXXX/oauth2/v2.0/authorize
  - application_secret=XXXX
  - skip_auth_uri=
  - namespace=istio-system
  - userid-header=kubeflow-userid
  - userid-prefix=

**Issue**
When using KFP client to upload the pipeline (client.pipeline_uploads.upload_pipeline()) with below client config throws an error. 

 `client = kfp.Client(host='https://<LoadBalancer IP Address>/pipeline', existing_token=<token>)`

Error 
HTTPSConnectionPool(host='<Host_IP>', port=443): Max retries exceeded with url: /pipeline/apis/v1beta1/pipelines/upload?name=local_exp-6714175b-6d59-40d0-9019-5b4ee58dc483 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1076)')))

Is there a way to override cert verification?

or

When using KFP client to upload the pipeline (client.pipeline_uploads.upload_pipeline()) with below client config redirects to google auth error. 

`client = kfp.Client(host='https://<LoadBalancer IP Address>/pipeline ,client_id=<client_id>, other_client_id=<client_id>,other_client_secret=<application_secret>,namespace='kfauth') `

image
   

**Environment:**
- Kubeflow version:  `v1.1.O`
- kfctl version: `kfctl_v1.1.0-0-g9a3621e_linux.tar.gz`
- kfp version: `1.0.1`
- python version: `3.6.8`
- kfp-server-api version: `1.0.1`
- Kubernetes platform: `Azure Kubernetes Service` 
- Kubernetes version: `1.17.11`

CC: @Bobgy

### PR DESCRIPTION
**Pull request by vmallya-123**
**Description of your changes:
Adds the option to override cert check
Fixes: #4569

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository]

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/ok-to-test

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -156,10 +158,9 @@ def __init__(self,
             KF_PIPELINES_APP_OAUTH2_CLIENT_ID_ENV)
         other_client_secret = other_client_secret or os.environ.get(
             KF_PIPELINES_APP_OAUTH2_CLIENT_SECRET_ENV)
-
         config = self._load_config(host, client_id, namespace, other_client_id,
                                    other_client_secret, existing_token, proxy,
-                                   ssl_ca_cert, kube_context, credentials)
+                                   ssl_ca_cert, kube_context, credentials,verify_ssl)
```
Comment (chensun) [inline sdk/python/kfp/_client.py:163]: nit: space

Please format the code following the contributing guide: https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#sdk

File: sdk/python/kfp/_client.py [hunk 2]
```diff
@@ -119,6 +119,7 @@ class Client(object):
           server.
       ui_host: Base url to use to open the Kubeflow Pipelines UI. This is used when running the client from a notebook to generate and
           print links.
+      verify_ssl: A boolean indication to verify the servers TLS certificate or not
```
Comment (chensun) [inline sdk/python/kfp/_client.py:122]: nit:  terminated by a period

**Review by chensun — COMMENTED** [review]
@vmallya-123 Thanks for your change. Can you please sign the CLA?

**Comment by chensun** [comment]
The following contributors were found for this pull request:

✅ ebbd5ee PR Opener: @vmallya-123
❌ ebbd5ee Author: <“v********lya​@tech.jago.com”>

@vmallya-123 Looks like you used a different email than the one you signed CLA?

**Comment by chensun** [comment]
Other than the CLA issue, can you please add an item in the SDK release note: https://github.com/kubeflow/pipelines/blob/master/sdk/RELEASE.md#bug-fixes-and-other-changes

**Comment by vmallya-123** [comment]
Hi @chensun the agreement was signed yesterday by my employer. I have also ensured that 
1) my email is added to my github account
2) keep my emails private is ticked off
3) owner of the google group for which the agreement was signed
4) signed an individual agreement too for this mail Author: <“v********lya​@tech.jago.com”>

i can see the cla data for <“v********lya​@tech.jago.com”>  when i check for details.

is there anything else that I should check for or do?

**Comment by chensun** [comment]
Hi @vmallya-123 , I rerun the CLA scan, and it still complains the same error

<h4 style="font-size: 13px; font-weight: bold; font-family: &quot;open sans&quot;, arial, sans-serif; margin: 1.236em 0px 0.618em; color: rgb(68, 68, 68); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Contributor(s)</h4>

GitHub Login | Email | State
-- | -- | --
vmallya-123 |   | Ok
  | “v********lya​@tech.jago.com” | Need CLA

And I pulled your PR locally, and I can see that your commits was made using the same email address as the CLA tool complained above. Can you please double check your commit details and see if there's a chance you misspell the email address in your commit?

**Comment by chensun** [comment]
> keep my emails private is ticked off

If that's your intention, don't forget to set your git commit email using the same norely.github.com email address
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git

**Comment by vmallya-123** [comment]
@chensun yes it was a typo. Sorry for the confusion and thanks :-)

**Comment by chensun** [comment]
/lgtm
/approve

Thanks!

**Comment by chensun** [comment]
/retest


kubeflow-pipelines-samples-v2 failure is unrelated, I'll ignore it if that's the only failed test.

**Comment by thesuperzapper** [comment]
@chensun is there any way we can back-port this into the 1.8 line of the SDK (as self-signed certificates are relatively common for testing environments).

**Pull request merged**